### PR TITLE
Minor logic simplifications and proper check for console

### PR DESCRIPTION
**Pull request by megawac**

### CODE REVIEW

File: src/router.js [hunk 1]
```diff
@@ -94,8 +93,7 @@ var ObjectRouter = BaseRouter.extend({
   // route for per-route handling, or override this method
   // to modify the default handling of errors.
   onError(e, routeData) {
-    if (!console) { return this; }
-    console.assert(false, e, e.stack);
+    if (typeof console === 'object') { console.assert(false, e, e.stack); }
```
Comment (megawac) [inline src/router.js]: Any reason you're using `console.assert` instead of `console.error`?

Comment (jmeas) [inline src/router.js]: Put this on 3 lines?

```js
if () {
  // stuff
}
```

Comment (jmeas) [inline src/router.js]: @megawac, nope. I just borrowed it from some code I saw in Ember / one of Tilde's libraries sometime. Feel free to change it!

Comment (megawac) [inline src/router.js]: Weird they were using it at all

**Comment by jmeas** [comment]
:+1:

Thanks, @megawac!

**Pull request merged**